core/runtime/op_driver/futures_unordered_driver.rs

```diff
@@ -0,0 +1,349 @@
+pub struct FuturesUnorderedDriver<
+  C: OpMappingContext + 'static = V8OpMappingContext,
+> {
+  len: Cell<usize>,
+  task: Cell<MaybeTask>,
+  task_set: Cell<bool>,
+  queue: SubmissionQueue<FuturesUnordered<FutureAllocation<PendingOp<C>>>>,
+  completed_ops: Rc<RefCell<VecDeque<PendingOp<C>>>>,
+  completed_waker: Rc<UnsyncWaker>,
+  arena: FutureArena,
+}
+
+impl<C: OpMappingContext + 'static> Drop for FuturesUnorderedDriver<C> {
+  fn drop(&mut self) {
+    if let MaybeTask::Handle(h) = self.task.take() {
+      h.abort()
+    }
+  }
+}
+
+impl<C: OpMappingContext> Default for FuturesUnorderedDriver<C> {
+  fn default() -> Self {
+    let (queue, results) = new_submission_queue();
+    let completed_ops = Rc::new(RefCell::new(VecDeque::with_capacity(128)));
+    let completed_waker = Rc::new(UnsyncWaker::default());
+    let task = MaybeTask::Task(Box::pin(poll_task(
+      results,
+      completed_ops.clone(),
+      completed_waker.clone(),
+    )))
+    .into();
+
+    Self {
+      len: Default::default(),
+      task,
+      task_set: Default::default(),
+      completed_ops,
+      queue,
+      completed_waker,
+      arena: Default::default(),
+    }
+  }
+}
+
+impl<C: OpMappingContext> FuturesUnorderedDriver<C> {
+  #[inline(always)]
+  fn ensure_task(&self) {
+    if !self.task_set.get() {
+      self.spawn_task();
+    }
+  }
+
+  #[inline(never)]
+  #[cold]
+  fn spawn_task(&self) {
+    let MaybeTask::Task(task) = self.task.replace(Default::default()) else {
+      unreachable!()
+    };
+    self.task.set(MaybeTask::Handle(spawn(task)));
+    self.task_set.set(true);
+  }
+
+  /// Spawn an unpolled task, along with a function that can map it to a [`PendingOp`].
+  #[inline(always)]
+  fn spawn_unpolled<R>(
+    &self,
+    task: impl Future<Output = R> + 'static,
+    map: impl FnOnce(R) -> PendingOp<C> + 'static,
+  ) {
+    self.ensure_task();
+    self.len.set(self.len.get() + 1);
+    self.queue.spawn(self.arena.allocate(task.map(map)));
+  }
+
+  /// Spawn a ready task that already has a [`PendingOp`].
+  #[inline(always)]
+  fn spawn_ready(&self, ready_op: PendingOp<C>) {
+    self.ensure_task();
+    self.len.set(self.len.get() + 1);
+    self.queue.spawn(self.arena.allocate(ready(ready_op)));
+  }
+
+  /// Spawn a polled task inside a [`FutureAllocation`], along with a function that can map it to a [`PendingOp`].
+  #[inline(always)]
+  fn spawn_polled<R>(
+    &self,
+    task: FutureAllocation<R>,
+    map: impl FnOnce(R) -> PendingOp<C> + 'static,
+  ) {
+    self.ensure_task();
+    self.len.set(self.len.get() + 1);
+    self.queue.spawn(self.arena.allocate(task.map(map)));
+  }
+
+  #[inline(always)]
+  fn pending_op_success<R: 'static>(
+    info: PendingOpInfo,
+    rv_map: C::MappingFn<R>,
+    v: R,
+  ) -> PendingOp<C> {
+    PendingOp(info, OpResult::new_value(v, rv_map))
+  }
+
+  #[inline(always)]
+  fn pending_op_failure<E: Into<Error> + 'static>(
```

Comment:
Are these two methods just to satisfy a trait? This seems super verbose, instead of inlining in `pending_op_result`. Maybe they're here to enforce types?

Reply:
It's a bit verbose, I agree. I removed OpError from the driver which makes these kind of redundant now. The previous code was actually a little hairy and repetitive. I'll fix before landing.

Reply:
Thank you

Reply:
Fixed